src/shell/dotnet/Shell/App.xaml.cs

```diff
@@ -175,4 +179,25 @@ private void InjectMessageRouterConfig()
                     
                 """);
     }
+   
+    private void InjectFDC3()
+    {
+        string bundle = @"fdc3-iife-bundle.js";
+        string iife = ReadResource(bundle);
+
+        WebWindow.AddPreloadScript(iife);
+    }
+
+    public string ReadResource(string name)
+    {
+        var assembly = Assembly.GetExecutingAssembly();    
+        string resourcePath = assembly.GetManifestResourceNames()
+            .Single(str => str.EndsWith(name));
```

Comment:
This doesn't seem very safe, it at the minimum adds an implicit requirement to have our resources not be postfixes of each other.